feat(config-conventional): footer/body-max-line

Fetz opened the pull request:
## Description
- Updated the config-conventional with footer/body max length
- Added tests to all the rules of config-conventional
- Updated the documentation of config-conventional to include missing rules

## Motivation and Context
Related to: https://github.com/marionebl/commitlint/issues/303#issuecomment-418525320

## Usage examples

```sh
# footer-line-max-length
echo "fix: some message
 BREAKING CHANGE: footer with multiple lines
has a message that is way too long and will break the line rule 'line-max-length' by several characters" # fails

 echo "fix: some message
 BREAKING CHANGE: footer with multiple lines
but still no line is too long" # passes
```

```sh
# body-line-max-length
echo "fix: some message
 body with multiple lines
has a message that is way too long and will break the line rule 'line-max-length' by several characters" # fails

echo "fix: some message
 body with multiple lines
but still no line is too long" # passes
```

## How Has This Been Tested?
- [x] Added tests
- [ ] Found a bug with the max-line-length when the commit message includes the git diff

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

marionebl commented:
@Fetz Are still interested in contributing this change?

Fetz commented:
Hey @marionebl this was blocked since I was getting this error:
https://github.com/conventional-changelog/commitlint/issues/437

and you recommended that the fix should be done on [conventional-commits-parser](https://github.com/conventional-changelog/conventional-changelog/tree/master/packages/conventional-commits-parser).

So I created a [ticket](https://github.com/conventional-changelog/conventional-changelog/issues/372) and I proposed some solutions and I asked for feedback on what solution should be approached and I even suggested to fix it, but I didn't get any feedback from any of the members...

marionebl commented:
> PS: now that you are a member of conventional-changelog maybe you can see if any of my suggestions are good to fix the issue on ticket so I can unblock this PR.

Ok, thanks for reestablishing the context for me. Let's take the discussion to the conventional-changelog issue then

marionebl commented:
Also - thanks for your patience 🙃. I investigated the conventional-changelog issue a bit. Thinking it through I'd sat we can move forward with this and resolve the `--verbose` back independently.

marionebl commented:
Thanks for your patience, btw 🙃. I investigated the parser issue a bit.

The scope of it is limited to commits read from COMMIT_EDIT_MSG created via --verbose, right?

Assuming this is true let's move forward with this feature and resolve the --verbose issue independently.

Fetz commented:
> The scope of it is limited to commits read from COMMIT_EDIT_MSG created via --verbose, right?

Yes it's true :)

>  let's move forward with this feature and resolve the --verbose issue independently.

Ok I will resolve this week with the conflicts so this PR is free to be merged

Fetz commented:
@marionebl  I updated the PR:
- rebased the PR with the latest changes
- updated the commit to use jest instead of ava

Note: 
- I saw some inconsistency between max length between the header, body and footer (since is only the header showing the current size of the header, should I update those in this PR or another?
- I saw the v12 tests failed (but it was with the format) maybe is a flaky test?

marionebl commented:
> I saw the v12 tests failed (but it was with the format) maybe is a flaky test?

That was `prettier` failing due to formatting inconsistencies, pushed the fix

> I saw some inconsistency between max length between the header, body and footer (since is only the header showing the current size of the header, should I update those in this PR or another?

Separate PR is fine
